Lib/test/test_frame.py

```diff
@@ -235,6 +235,27 @@ def inner():
                          r"^<frame at 0x[0-9a-fA-F]+, file %s, line %d, code inner>$"
                          % (file_repr, offset + 5))
 
+class TestIncompleteFrameAreInvisible(unittest.TestCase):
+
+    def test_issue95818(self):
+        #See GH-95818 for details
+        import gc
+        gc.set_threshold(1,1,1)
```

Comment:
This would alter gc threshold for all other tests, Can you reset it with a finally block and `gc.get_threshold`?

Reply:
`self.addCleanup(gc.set_threshold, *gc.get_threshold())`

https://github.com/python/cpython/blob/884eba3c76916889fd6bff3b37b8552bfb4f9566/Lib/test/test_weakset.py#L428